docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+    - Each service will have configuration that specifies a message endpoint for the service metrics.  The metrics message topic communications may be secured or unsecured (just as application services provide the means to export to secured or unsecured message pipes today).
+- All services must document what metrics they offer.
+- All EdgeX services must implement a common metrics interface/contract that defines an API set about service metrics.
+    - The metrics REST endpoints implemented on each service are not there to provide the metrics data, but to know what metrics the service provides, to know the the current state for each metric (`on` or `off`), and to provide a means to turn `on` or `off` the metrics collection.  The actual metric data will be provided, when collected, in messages to a message bus.
+    - The interface would include the definition of a GET REST endpoint that responds with the metrics that the service offers (and whether that metric is currently `on` or `off`). The proposed endpoint format:  **/api/v2/telemetry**.
+    - The interface would define PUT and PATCH REST endpoints that allows the user to toggle between `on` and `off` for any metric.  The proposed endpoint format:  **/api/v2/telemetry**.  In this REST request body would be the list of metric names to be turned `on` (or those turned `on` or `off` in the case of PATCH).  It is assumed that those metrics not listed are to be turned `off` in the PUT call where PATCH request is explicit about which metrics are turned `on` or `off` and leaves others unchanged.
+- Services will have configuration which allows EdgeX system managers to select which metrics are `on` or `off` by default - in other words providing the initial bootstrapping configuration that determines what metrics are collected and reported by default.
+    - When a metric is turned `off` the service does not report the metric.  When a metric is turned `on` the service collects and sends the metric to the designated message topic.
+    - Per REST API described above, the `on` and `off` control of the metrics collected can be changed during runtime of the service.
+    - Metrics collection must be pushed to the designated message topic on some appointed schedule.  The schedule would be designated by configuration in the schedule service or done in a way similar to auto events in device services.  
+        - *Do we want to dictate this or allow services to implement as they see fit?*
+            - Recommendation by @cloudxxx8 - have internal scheduler but provide option to use external scheduler.   The service then has to provide the API for the external to call for collection.  Suggest using a design similar to auto discovery in device services.
+        - *Also, should there be an interval per metric or a single interval for all metrics collected and pushed?*
+            - Recommendation by @cloudxxx8 to support interval per metric but to have one for all as default
+

```

Comment:
Scheduling seems a bit hand-wavy here.

If each metric only supports a boolean status, it's not clear what the behavior is when a metric is set to `on` and no scheduling has been configured?

In order to use the support-scheduler, we'd need to implement a REST API trigger which has been mentioned, but isn't called out as a requirement in this version of the ADR.

If we were to support an internal auto-event like scheduler, then my suggestion would be to make this part of the per-metric configuration, and I'd like to see it actually described.